internal/scheduler/database/job_repository.go

```diff
@@ -72,20 +73,48 @@ func NewPostgresJobRepository(db *pgxpool.Pool, batchSize int32) *PostgresJobRep
 // FetchJobRunErrors returns all armadaevents.JobRunErrors for the provided job run ids.  The returned map is
 // keyed by job run id.  Any dbRuns which don't have errors wil be absent from the map.
 func (r *PostgresJobRepository) FetchJobRunErrors(ctx context.Context, runIds []uuid.UUID) (map[uuid.UUID]*armadaevents.Error, error) {
-	queries := New(r.db)
-	rows, err := queries.SelectRunErrorsById(ctx, runIds)
-	if err != nil {
-		return nil, errors.WithStack(err)
-	}
-	errorsByRunId := make(map[uuid.UUID]*armadaevents.Error, len(rows))
+	chunks := armadaslices.Partition(runIds, int(r.batchSize))
+
+	errorsByRunId := make(map[uuid.UUID]*armadaevents.Error, len(runIds))
 	decompressor := compress.NewZlibDecompressor()
-	for _, row := range rows {
-		jobError, err := protoutil.DecompressAndUnmarshall(row.Error, &armadaevents.Error{}, decompressor)
-		if err != nil {
-			return nil, errors.WithStack(err)
+
+	err := r.db.BeginTxFunc(ctx, pgx.TxOptions{
+		IsoLevel:       pgx.ReadCommitted,
+		AccessMode:     pgx.ReadWrite,
+		DeferrableMode: pgx.Deferrable,
+	}, func(tx pgx.Tx) error {
+		for _, chunk := range chunks {
+			tmpTable, err := insertRunIdsToTmpTable(ctx, tx, chunk)
```

Comment:
What is the advantage of doing it this was vs just calling `SelectRunErrorsById` in chunks?

Reply:
`SelectRunErrorsById` is using an `IN` clause.  Really large`IN` Clauses (think tens of thousands of values) tend not to be very efficient and extremely large `IN` Clauses  don't work at all.  Using a tmp table means you can have larger chunk sizes and should always get decent performance.

There is a question as to whether we could just cap the chunk size at 10K and test to see if postgres `IN` is just as fast (or faster) than the tmp table approach at this scale.  If that's the case then we could probably get away  with the simpler approach.

Reply:
Sounds like a reasonable explanation - was just trying to avoid code complicated if we didn't need it

But sounds fair - especially if we want to handle large number of ids (and hence large chunk sizes)

I'll approve for now - if you want to test just having a 10k chunk size and doing an `IN` then that is up to you